src/run/parser/time_split_tracker.rs

```diff
@@ -124,7 +125,21 @@ pub fn parse<R: BufRead>(source: R, path_for_loading_other_files: Option<PathBuf
     parse_history(&mut run, path).ok();
 
     for comparison in comparisons {
-        run.add_custom_comparison(comparison);
+        let mut name = comparison.to_string();
+        let mut number = 0;
+        loop {
+            match run.add_custom_comparison(name.clone()) {
```

Comment:
There's multiple problems with this approach. First, you can match on Err(ComparisonError::DuplicateName) instead of the ref e with if. Then you don't need the impossible case.

Reply:
Second, I noticed that we do this here way too late. The segments with the comparison times have already been added at this point. So there's a mismatch every time there's a rename like that. So you probably want to actually add these comparisons early (where we create the comparisons list up there) and do the inserting and renaming logic up there.

Reply:
Also the number renaming seems mostly fine to me. I'd probably start with 2 instead of 0 though ("Bla", "Bla2", "Bla3").

Reply:
I'm not sure if I like the `[ok]` though. I'd probably just cut away the [Race] and then try using that.

Reply:
Also I just noticed that in the duplicate name case, you actually keep adding the numbers, so it's ("Bla", "Bla2", "Bla23", "Bla234", ...), which is probably not what we want.